src/components/layout.tsx

```diff
@@ -36,7 +36,7 @@ const Layout = ({ children, title, description, img }: Props) => {
   const onIntersectionChange = (entries: IntersectionObserverEntry[]) => {
     entries.forEach((entry: IntersectionObserverEntry) => {
       const element = entry.target as HTMLElement;
-      element.style.color = entry.isIntersecting ? '#000' : '#fff';
+      element.style.color = entry.intersectionRatio > 0.5 ? '#000' : '#fff';
```

Comment:
nice 👍 while we're at it, can we add a `transition: color .28s` to the navigation titles to make the color tween less jarring?

Reply:
Done. Looks so much smoother now! Thanks! 💯